tm2/pkg/sdk/baseapp.go

```diff
@@ -652,25 +652,34 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 		// Each message result's Data must be length prefixed in order to separate
 		// each result.
 		data = append(data, msgResult.Data...)
-		events = append(events, msgResult.Events...)
-		defer func() {
-			events = append(events, ctx.EventLogger().Events()...)
-			result.Events = events
-		}()
-		// TODO append msgevent from ctx. XXX XXX
+
+		// each msgs' event
+		var msgEvent []abci.Event
```

Comment:
You know at this point in time how many events you'll have, so this slice should be initialized with a capacity:
```go
msgEvent := make([]abci.Event, 0, len(msgCtxEvents))
```